framework/src/application.ts

```diff
@@ -203,9 +224,28 @@ export class Application {
 		application._registerModule(tokenModule);
 		application._registerModule(feeModule);
 		application._registerModule(interoperabilityModule);
-		application._registerModule(posModule);
+		if (usePoA) {
+			application._registerModule(poaModule);
+		} else {
+			application._registerModule(posModule);
+		}
 		application._registerModule(randomModule);
-		application._registerModule(dynamicRewardModule);
+
+		if (usePoA) {
+			return {
+				app: application,
+				method: {
+					validator: validatorModule.method,
+					token: tokenModule.method,
+					auth: authModule.method,
+					fee: feeModule.method,
+					poa: poaModule.method,
+					random: randomModule.method,
+					reward: rewardModule.method,
+					interoperability: interoperabilityModule.method,
+				},
+			};
+		}
```

Comment:
we should not touch this file, instead what we did here for PoA, we simply do it under `src/app/app.ts` in `poa-mainchain`

Reply:
If that's the case do I need to de-register pos from defaultApplication?